ProcessMaker/processmaker
# FOUR-9551 | Front End: Add Project Select List to Each Asset Type Within the Designer

# User Story
Tickets: 
[FOUR-9551](https://processmaker.atlassian.net/browse/FOUR-9551)
[FOUR-9559](https://processmaker.atlassian.net/browse/FOUR-9559)

As a Solution Developer, I want to assign any asset I create to a Project so that I can contribute to my solution design as a part of my regular asset development.

At present, the asset ellipsis menu, located within each asset's listing page under the designer tab, lacks an important feature - the ability to assign an asset to one or more projects simultaneously. To address this limitation, we aim to introduce a new action named 'Assign to Project' in the asset ellipsis menu. When users select this action, a popup should appear, presenting them with the option to choose multiple projects to which they can add the asset.

Also, when the "+ [[Asset Type]]" button is clicked, the subsequent modal allows users to add an asset but lacks the ability to associate it with a specific project. The objective of this ticket is to enhance the modal to include a project selection option.

# Work Done
1. Introduce a new action, 'Assign to Project', in the asset ellipsis menu, enabling users to associate the asset with one or more projects.
2. Upon selecting the 'Assign to Project' action, display a user-friendly popup that allows users to conveniently select multiple projects for adding the asset.
3. Add a new option to the "Add Asset" modal and asset configuration pages, enabling users to select a project the asset will be associated with.
4. Implement a query to populate the select list with existing Projects, ensuring that users can choose from the available options.

# How to Test FOUR-9559
1. Go to branch `feature/FOUR-9551` in `processmaker`.
2. Go to branch `feature/FOUR-9551` in `package-projects`. Install the package.
3. Go to branch `feature/FOUR-9551` in `package-decision-engine`. Install the package.
4. Go to branch `feature/FOUR-9551` in `package-data-sources`. Install the package.
5. Go to Designer → Projects → +ADD PROJECT. Create at least two projects if you don't already have projects created in your environment.
6. Go to Designer → Processes → Ellipsis Menu → `Assign to Project`.
	- A pop-up should appear with an option to select a Project.
	- The select list should include the available Projects in your environment.
	- You should be able to select multiple Projects.
7. Repeat Step 6 for Screens, Scripts, Data Connectors, and Decision Tables.

# How to Test FOUR-9551
1. Ensure you are on the same branches you were on when testing FOUR-9559.
2. Ensure you have projects created in your environment.
3. Go to Designer → Processes → +PROCESS. Select a template to use or select 'Build Your Own'.
	- An option to select a Project should appear in the Create Process modal.
		- The select list should include the available Projects in your environment.
		- You should be able to select multiple Projects.
4. Go to Designer → Processes → Configure.
	- An option to select a Project should appear in the Process Configuration tab.
		- The select list should include the available Projects in your environment.
		- You should be able to select multiple Projects.
5. Repeat Steps 3 and 4 for Screens, Scripts, Data Connectors, and Decision Tables.


ci:package-projects:feature/FOUR-9551
ci:package-data-sources:feature/FOUR-9551
ci:package-decision-engine:feature/FOUR-9551
ci:deploy

# Code Review Checklist


[FOUR-9551]: https://processmaker.atlassian.net/browse/FOUR-9551?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[FOUR-9559]: https://processmaker.atlassian.net/browse/FOUR-9559?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
